### ISSUE TITLE
cross-signing: my server can't see cross signing keys of Ben

### ISSUE THREAD

**Issue by MatMaul** [matrix-org/synapse#7521]
When I trigger a verification the old way is used with Ben, he can see a modal dialog.
We are both on 1.13.0rc2, and it seems to happen whatever the client we used.

It seems that my server can't see cross-signing identity of Ben at all, here is what I get when I open the side bar for Ben user:
`got cross-signing keys for @ben:banzan.uk: {master: undefined, self_signing: undefined, user_signing: undefined}`

**Comment by anoadragon453** [comment]
When viewing `@ben:banzan.uk` profile in Riot I get `got cross-signing keys for @ben:banzan.uk: Object { master: {…}, self_signing: {…}, user_signing: undefined }`.

  This is a result of a request to `POST /_matrix/client/r0/keys/query` with the body:

  ```
  {
    "device_keys": {
      "@ben:banzan.uk": []
    },
    "token": "s8181613_217014825_78158_12649083_676265_886_238619_1632516_130"
  }
  ```

  with the response: https://gist.github.com/anoadragon453/943244e01cd8ca062160b4f1261f042a.

> When I trigger a verification the old way is used with Ben, he can see a modal dialog.

I'm not sure what this means. What does the dialog contain, if anything?

Some server logs from your server could help here as well. grepping for `ERROR ` around the time you view ben's profile, or any outbound requests you make that return a non-200 error code.

**Comment by MatMaul** [comment]
By looking at the network analyser I can confirm that the `/keys/query` request has empty master_keys and self_signing_keys for Ben on my server, unlike what I can see in your gist.
Nothing appears on my server logs :( I just restarted with 1.13.0rc3, the only error message I got was an unrelated `ERROR - PUT-2333 - Failed to handle edu 'm.receipt'`.
> I'm not sure what this means. What does the dialog contain, if anything?
It's on @benbz side, but I think what he meant that he get a popup window with emojis verification like when doing old school device-device verification, and not the new verification flow that happen in the side bar now I think ? Unsure about this one, I remember getting a popup window for verification even with cross signing (with QR Code possible) so it is perhaps irrelevant.

**Comment by benbz** [comment]
Sorry, only just getting around to this now.

The background to this is that we were on a cross-signing jitsi and, after cross-signing someone else, @MatMaul & I attempted to cross sign. I commented that it was odd that the emojis for the 1st person were in the sidebar, but the emojis for @MatMaul were in a modal dialogue. @t3chguy pointed out that this was the legacy verification flow not the cross-signing one.

Everyone else we both attempted to verify used the emoji in the sidebar / cross-signing flow. Every time we tried to verify each other it used the legacy flow.

**Comment by MatMaul** [comment]
It looks the same as #7504 after checking.

**Comment by MatMaul** [comment]
We both updated to 1.14.0rc1, unfortunately no luck :/ I haven't try any of the SQL queries in #7504 ticket, but it would be nice if it is solved for all users at some point, so we are keeping it like that for now, feel free to use us as guinea pigs.

**Comment by babolivier** [comment]
Should be fixed by #7594 (+ inserting rows à la #7504)
